File: 2-year/Java/Lab1/src/main/java/com/Starygin/Presentation/Scenarios/GetBalanceScenario.java
package com.Starygin.Presentation.Scenarios;

import com.Starygin.Application.Contracts.UserService;
import com.Starygin.Presentation.Scenario;

import java.io.IOException;
import java.util.Scanner;

public class GetBalanceScenario implements Scenario {
    private final UserService userService;

    public GetBalanceScenario(UserService userService) {
        this.userService = userService;
    }

    @Override
    public String getName() {
        return "Get Balance";
    }

    @Override
    public void run() {
        System.out.println("Enter account id: ");
        Scanner scanner = new Scanner(System.in);

        try {
            long accountId = scanner.nextLong();

            double balance = userService.getBalance(accountId);

            System.out.println("Your balance is: " + balance);

        } catch (IOException e) {
            System.out.println(e.getMessage());
        }
    }
}

File: 2-year/OOP-CSharp/src/Lab5/Presentation/Lab5.Presentation.Console/Scenarios/GetBalance/GetBalanceScenario.cs
using Lab5.Application.Contracts.Users;
using Lab5.Application.Contracts.Users.OperationsResults;
using Spectre.Console;
using System.Globalization;

namespace Lab5.Presentation.Console.Scenarios.GetBalance;

public class GetBalanceScenario : IScenario
{
    private readonly IUserService _userService;

    public GetBalanceScenario(IUserService userService)
    {
        _userService = userService;
    }

    public string Name => "Get Balance";

    public void Run()
    {
        long accountId = AnsiConsole.Ask<long>("Enter your account id: ");

        GetBalanceResult result = _userService.GetBalance(accountId);

        string message = result switch
        {
            GetBalanceResult.Success success => success.Balance.ToString(CultureInfo.CurrentCulture),
            GetBalanceResult.AccountNotFound => "Account not found",
            GetBalanceResult.NotLoginYet => "Please login to continue",
            _ => throw new ArgumentOutOfRangeException(nameof(result)),
        };

        AnsiConsole.WriteLine(message);
        AnsiConsole.Ask<string>("Ok");
    }
}